### ISSUE TITLE
replies missing content and in-reply-to

### ISSUE THREAD

**Issue by karadaisy** [snarfed/ownyourresponses#1]
I remembered and/or figured out what I was trying to explain earlier -- backfed replies are missing `content` and `in-reply-to`.

Little snippet of testing code

``` python
twitter = gr_twitter.Twitter(access_token_key, access_token_secret)
activities = twitter.get_activities(activity_id='651630562050899969')
mf2json  = [gr_mf2.object_to_json(a) for a in activities]
```

produces activity:

```
[{'actor': {'description': u'I the undersigned shall forfeit all rights, privileges and licenses herein and herein contained\u2026 Fax mentis incendium gloria cultum\u2026 Memo bis punitor delicatum!',
            'displayName': u'Kyle Mahan',
            'id': u'tag:twitter.com:kylewmahan',
            'image': {'url': u'https://twitter.com/kylewmahan/profile_image?size=original'},
            'location': {'displayName': u'Burlingame, CA'},
            'numeric_id': u'19090770',
            'objectType': 'person',
            'published': '2009-01-16T22:51:18+00:00',
            'url': u'https://kylewm.com',
            'urls': [{'value': u'https://kylewm.com'}],
            'username': u'kylewmahan'},
  'context': {'inReplyTo': [{'id': 'tag:twitter.com:651619881318678530',
                             'url': u'https://twitter.com/DaveCA/status/651619881318678530'}]},
  'generator': {'displayName': u'Twitter Web Client',
                'url': u'http://twitter.com'},
  'id': u'tag:twitter.com:651630562050899969',
  'object': {'author': {'description': u'I the undersigned shall forfeit all rights, privileges and licenses herein and herein contained\u2026 Fax mentis incendium gloria cultum\u2026 Memo bis punitor delicatum!',
                        'displayName': u'Kyle Mahan',
                        'id': u'tag:twitter.com:kylewmahan',
                        'image': {'url': u'https://twitter.com/kylewmahan/profile_image?size=original'},
                        'location': {'displayName': u'Burlingame, CA'},
                        'numeric_id': u'19090770',
                        'objectType': 'person',
                        'published': '2009-01-16T22:51:18+00:00',
                        'url': u'https://kylewm.com',
                        'urls': [{'value': u'https://kylewm.com'}],
                        'username': u'kylewmahan'},
             'content': u"@DaveCA @mcbobo86 one time i had leftover ratatouille, put it over pan fried potatoes, and a poached egg on top. best breakfast i've made",
             'id': u'tag:twitter.com:651630562050899969',
             'inReplyTo': [{'id': 'tag:twitter.com:651619881318678530',
                            'url': u'https://twitter.com/DaveCA/status/651619881318678530'}],
             'objectType': 'note',
             'published': '2015-10-07T05:30:07+00:00',
             'tags': [{'displayName': u'DaveCA',
                       'id': u'tag:twitter.com:DaveCA',
                       'length': 7,
                       'objectType': 'person',
                       'startIndex': 0,
                       'url': u'https://twitter.com/DaveCA'},
                      {'displayName': u'Megan',
                       'id': u'tag:twitter.com:mcbobo86',
                       'length': 9,
                       'objectType': 'person',
                       'startIndex': 8,
                       'url': u'https://twitter.com/mcbobo86'}],
             'to': [{'alias': '@public', 'objectType': 'group'}],
             'url': u'https://twitter.com/kylewmahan/status/651630562050899969'},
  'published': '2015-10-07T05:30:07+00:00',
  'url': u'https://twitter.com/kylewmahan/status/651630562050899969',
  'verb': 'post'}]
```

and a paltry mf2 object:

```
[{'properties': {'author': [{'properties': {'location': [{'properties': {'name': [u'Burlingame, CA']},
                                                          'type': ['h-card',
                                                                   'p-location']}],
                                            'name': [u'Kyle Mahan'],
                                            'photo': [u'https://twitter.com/kylewmahan/profile_image?size=original'],
                                            'published': ['2009-01-16T22:51:18+00:00'],
                                            'uid': [u'tag:twitter.com:kylewmahan'],
                                            'url': [u'https://kylewm.com']},
                             'type': ['h-card']}],
                 'published': ['2015-10-07T05:30:07+00:00'],
                 'uid': [u'tag:twitter.com:651630562050899969'],
                 'url': [u'https://twitter.com/kylewmahan/status/651630562050899969']},
  'type': ['h-entry']}]
```

**Comment by snarfed** [comment]
huh. that resulting mf2 object without content etc is surprising, and definitely bad and should be fixed.

... and heh, i was about to say this should maybe be a granary bug instead, since ownyourresponses explicitly generates content and in-reply-to and passes them in the micropub request, and i started to find example complete replies it's sent me...

which led me to realize that https://snarfed.org/responses shows recent RTs and favorites from oyr but no recent replies.

so i think you're right, replies are just broken full stop. whee!

**Comment by karadaisy** [comment]
let's track here https://github.com/snarfed/granary/issues/44

**Comment by snarfed** [comment]
reopening just for tracking. we still think the fix will come entirely from snarfed/granary#44.

**Comment by snarfed** [comment]
tentatively closing because snarfed/granary#44 is now done.
